File: plugins/scandit-sdk/evals/id-capture-net-android/fixtures/IntegratedActivity.cs
using Android.OS;
using Android.Views;
using Android.Widget;

using Scandit.DataCapture.Core.Capture;
using Scandit.DataCapture.Core.Source;
using Scandit.DataCapture.Core.UI;
using Scandit.DataCapture.ID.Capture;
using Scandit.DataCapture.ID.Data;
using Scandit.DataCapture.ID.UI.Overlay;

namespace MyApp;

[Activity(Label = "@string/app_name", MainLauncher = true, Theme = "@style/Theme.AppCompat.Light.NoActionBar")]
public class MainActivity : CameraPermissionActivity, IIdCaptureListener
{
    public const string SCANDIT_LICENSE_KEY = "-- ENTER YOUR SCANDIT LICENSE KEY HERE --";

    private DataCaptureContext dataCaptureContext = null!;
    private Camera? camera;
    private IdCapture idCapture = null!;
    private DataCaptureView dataCaptureView = null!;
    private IdCaptureOverlay overlay = null!;

    protected override void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);

        this.dataCaptureContext = DataCaptureContext.ForLicenseKey(SCANDIT_LICENSE_KEY);

        this.camera = Camera.GetDefaultCamera();
        if (this.camera is not null)
        {
            _ = this.camera.ApplySettingsAsync(IdCapture.RecommendedCameraSettings);
            _ = this.dataCaptureContext.SetFrameSourceAsync(this.camera);
        }

        var settings = new IdCaptureSettings
        {
            AcceptedDocuments =
            [
                new Passport(IdCaptureRegion.Any),
                new DriverLicense(IdCaptureRegion.Any),
                new IdCard(IdCaptureRegion.Any),
            ],
            Scanner = new IdCaptureScanner(
                physicalDocument: new FullDocumentScanner(),
                mobileDocument: null),
        };

        this.idCapture = IdCapture.Create(this.dataCaptureContext, settings);

        var container = new FrameLayout(this);
        this.SetContentView(container);
        this.dataCaptureView = DataCaptureView.Create(this.dataCaptureContext);
        container.AddView(
            this.dataCaptureView,
            new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent));
        this.overlay = IdCaptureOverlay.Create(this.idCapture, this.dataCaptureView);
        this.overlay.IdLayoutStyle = IdLayoutStyle.Square;
    }

    protected override void OnResume()
    {
        base.OnResume();
        this.idCapture.AddListener(this);
        this.idCapture.Enabled = true;
        this.RequestCameraPermission();
    }

    protected override void OnPause()
    {
        base.OnPause();
        this.camera?.SwitchToDesiredStateAsync(FrameSourceState.Off);
        this.idCapture.RemoveListener(this);
    }

    protected override void OnCameraPermissionGranted()
    {
        this.camera?.SwitchToDesiredStateAsync(FrameSourceState.On);
    }

    protected override void OnDestroy()
    {
        base.OnDestroy();
        this.overlay.Dispose();
        this.idCapture.Dispose();
    }

    public void OnIdCaptured(IdCapture mode, CapturedId capturedId)
    {
        string? fullName = capturedId.FullName;
        DateResult? dateOfBirth = capturedId.DateOfBirth;
        string? documentNumber = capturedId.DocumentNumber;

        mode.Enabled = false;
        this.RunOnUiThread(() =>
        {
            // Present fullName / dateOfBirth / documentNumber.
        });
    }

    public void OnIdRejected(IdCapture mode, CapturedId? capturedId, RejectionReason reason)
    {
        mode.Enabled = false;
        this.RunOnUiThread(() =>
        {
            // Show a message based on `reason`.
        });
    }
}

File: plugins/scandit-sdk/evals/id-capture-android/fixtures/IntegratedActivity.kt
package com.example.myapp

import android.app.AlertDialog
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.scandit.datacapture.core.capture.DataCaptureContext
import com.scandit.datacapture.core.source.Camera
import com.scandit.datacapture.core.source.FrameSourceState
import com.scandit.datacapture.core.ui.DataCaptureView
import com.scandit.datacapture.id.capture.*
import com.scandit.datacapture.id.data.*
import com.scandit.datacapture.id.ui.overlay.IdCaptureOverlay

class IdCaptureActivity : AppCompatActivity(), IdCaptureListener {

    private val dataCaptureContext =
        DataCaptureContext.forLicenseKey("-- ENTER YOUR SCANDIT LICENSE KEY HERE --")

    private val camera = Camera.getDefaultCamera(IdCapture.createRecommendedCameraSettings())

    private val idCapture: IdCapture

    init {
        dataCaptureContext.setFrameSource(camera)

        val settings = IdCaptureSettings().apply {
            acceptedDocuments = listOf(
                IdCard(IdCaptureRegion.ANY),
                DriverLicense(IdCaptureRegion.ANY),
                Passport(IdCaptureRegion.ANY),
            )
            scanner = IdCaptureScanner(FullDocumentScanner())
        }

        idCapture = IdCapture.forDataCaptureContext(dataCaptureContext, settings)
        idCapture.addListener(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val dataCaptureView = DataCaptureView.newInstance(this, dataCaptureContext)
        IdCaptureOverlay.newInstance(idCapture, dataCaptureView)
        setContentView(dataCaptureView)
        // Request CAMERA permission here before scanning starts.
    }

    override fun onResume() {
        super.onResume()
        idCapture.isEnabled = true
        camera?.switchToDesiredState(FrameSourceState.ON)
    }

    override fun onPause() {
        idCapture.isEnabled = false
        camera?.switchToDesiredState(FrameSourceState.OFF)
        super.onPause()
    }

    override fun onDestroy() {
        idCapture.removeListener(this)
        dataCaptureContext.removeCurrentMode()
        super.onDestroy()
    }

    override fun onIdCaptured(mode: IdCapture, id: CapturedId) {
        mode.isEnabled = false
        val message = descriptionFor(id)
        runOnUiThread {
            showResultDialog("Recognized Document", message)
        }
    }

    override fun onIdRejected(mode: IdCapture, id: CapturedId?, reason: RejectionReason) {
        mode.isEnabled = false
        val message = if (reason == RejectionReason.TIMEOUT) {
            "Document capture failed. Make sure the document is well lit and free of glare."
        } else {
            "Document not supported. Try scanning another document."
        }
        runOnUiThread {
            showResultDialog("Rejected", message)
        }
    }

    private fun showResultDialog(title: String, message: String) {
        AlertDialog.Builder(this)
            .setTitle(title)
            .setMessage(message)
            .setPositiveButton("OK") { _, _ -> idCapture.isEnabled = true }
            .setOnCancelListener { idCapture.isEnabled = true }
            .show()
    }

    private fun descriptionFor(capturedId: CapturedId): String {
        val parts = mutableListOf<String>()
        capturedId.fullName?.let { parts.add("Name: $it") }
        capturedId.dateOfBirth?.let { parts.add("DOB: ${it.day}/${it.month}/${it.year}") }
        capturedId.dateOfExpiry?.let { parts.add("Expiry: ${it.day}/${it.month}/${it.year}") }
        capturedId.documentNumber?.let { parts.add("Doc #: $it") }
        capturedId.nationality?.let { parts.add("Nationality: $it") }
        capturedId.document?.documentType?.let { parts.add("Type: $it") }
        return parts.joinToString("\n")
    }
}
